scripts/install-macos.sh

```diff
@@ -0,0 +1,268 @@
+        OK=true;    
+    else
+        OK=false;
+    fi
+}
+
+function isGit() {
+    if git --version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isGo() {
+    if go version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+function isGlide() {
+    if glide --version; then
+        OK=true;
+    else
+        OK=false;
+    fi
+}
+
+# Installation functions begin here
+
+function installGo() {
+    echo 'Installing Golang on your machine.'
+    # Go install script based on https://github.com/canha/golang-tools-install-script
+    VERSION="1.15.7"
+
+    [ -z "$GOROOT" ] && GOROOT="$HOME/go/go-install-do-not-delete"
+    [ -z "$GOPATH" ] && GOPATH="$HOME/go"
+
+    OS="$(uname -s)"
+    ARCH="$(uname -m)"
+
+    case $OS in
+        "Linux")
+            case $ARCH in
+            "x86_64")
+                ARCH=amd64\
+                ;;
+            "aarch64")
+                ARCH=arm64
+                ;;
+            "armv6" | "armv7l")
+                ARCH=armv6l
+                ;;
+            "armv8")
+                ARCH=arm64
+                ;;
+            .*386.*)
+                ARCH=386
+                ;;
+            esac
+            PLATFORM="linux-$ARCH"
+        ;;
+        "Darwin")
+            PLATFORM="darwin-amd64"
+        ;;
+    esac
+
+    if [ -z "$PLATFORM" ]; then
+        echo "Your operating system is not supported by the script."
+        exit 1
+    fi
+
+    if [ '/bin/bash' == `which bash` ]; then 
+        shell_profile="$HOME/.bashrc"
+        echo "bash is installed"
+    else
+        echo "Kelp requires bash"
+    fi
+
+    {
+        echo '# GoLang'
+        echo "export GOROOT=${GOROOT}"
+        echo 'export PATH=$GOROOT/bin:$PATH'
+        echo "export GOPATH=$GOPATH"
+        echo 'export PATH=$GOPATH:$PATH'
+    } >> "$shell_profile"
+
+    if [ -d "$GOROOT" ]; then
+        echo "The Go install directory ($GOROOT) already exists."
+    fi
+
+    PACKAGE_NAME="go$VERSION.$PLATFORM.tar.gz"
+    TEMP_DIRECTORY=$(mktemp -d)
+
+    echo "Downloading $PACKAGE_NAME ..."
+    if hash wget 2>/dev/null; then
+        wget https://storage.googleapis.com/golang/$PACKAGE_NAME -O "$TEMP_DIRECTORY/go.tar.gz"
+    else
+        curl -o "$TEMP_DIRECTORY/go.tar.gz" https://storage.googleapis.com/golang/$PACKAGE_NAME
+    fi
+
+    if [ $? -ne 0 ]; then
+        echo "Download failed! Exiting."
+        exit 1
+    fi
+
+    echo "Extracting File..."
+
+    # sudo chown -R $USER: $HOME # https://github.com/golang/go/issues/27187
+    mkdir -p "$GOROOT"
+
+    tar -C "$GOROOT" --strip-components=1 -xzf "$TEMP_DIRECTORY/go.tar.gz"
+
+    # GOROOT=${GOROOT}
+    # PATH=$GOROOT/bin:$PATH
+    # GOPATH=$GOPATH
+    # PATH=$GOPATH:$PATH
+
+    mkdir -p "${GOPATH}/"{src,pkg,bin}
+    echo -e "\nGo $VERSION was installed into $GOROOT.\nMake sure to relogin into your shell or run:"
+    echo -e "\n\ restart script to update your environment variables."
```

Comment:
do we need them to restart the script to update the environment vars?

we can just do `source ~/.bashrc` to execute the `.bashrc` file instead